Exoft.Gamification.Api.Services/OrderService.cs

```diff
@@ -12,53 +11,36 @@
 using Exoft.Gamification.Api.Services.Interfaces;
 using Exoft.Gamification.Api.Services.Interfaces.Services;
 
-using File = Exoft.Gamification.Api.Data.Core.Entities.File;
-
 namespace Exoft.Gamification.Api.Services
 {
     public class OrderService : IOrderService
     {
         private readonly IOrderRepository _orderRepository;
         private readonly ICategoryRepository _categoryRepository;
-        private readonly IFileRepository _fileRepository;
+        private readonly IFileService _fileService;
         private readonly IUnitOfWork _unitOfWork;
         private readonly IMapper _mapper;
 
         public OrderService
         (
             IOrderRepository orderRepository,
             ICategoryRepository categoryRepository,
-            IFileRepository fileRepository,
+            IFileService fileService,
             IUnitOfWork unitOfWork,
             IMapper mapper
         )
         {
             _orderRepository = orderRepository;
             _categoryRepository = categoryRepository;
-            _fileRepository = fileRepository;
+            _fileService = fileService;
             _unitOfWork = unitOfWork;
             _mapper = mapper;
         }
 
         public async Task<ReadOrderModel> AddOrderAsync(CreateOrderModel model)
         {
             var order = _mapper.Map<Order>(model);
-
-            if (model.Icon != null)
-            {
-                using (var memory = new MemoryStream())
-                {
-                    await model.Icon.CopyToAsync(memory);
-
-                    var file = new File
-                    {
-                        Data = memory.ToArray(),
-                        ContentType = model.Icon.ContentType
-                    };
-                    await _fileRepository.AddAsync(file);
-                    order.IconId = file.Id;
-                }
-            }
+            order.IconId = (await _fileService.AddOrUpdateFileByIdAsync(model.Icon, order.IconId)).GetValueOrDefault();
```

Comment:
Why we need method GetValueOrDefault there?

Reply:
Because there IconId is not nullable, so i return Guid type by using this function